src/main/java/tasks/Deadline.java

```diff
@@ -0,0 +1,47 @@
+package tasks;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Represents a deadline task with a specific due date and time.
+ * A deadline task includes a description and a due date/time by which the task should be completed.
+ */
+public class Deadline extends Task {
+    protected LocalDateTime dueWhen;
+
+    /**
+     * Constructs a deadline task with the specified description and due date/time.
+     *
+     * @param description the description of the deadline task
+     * @param dueWhen the due date and time of the deadline task
+     */
+    public Deadline(String description, LocalDateTime dueWhen) {
+        super(description);
+        this.dueWhen = dueWhen;
+    }
+
+    /**
+     * Returns the string representation of the deadline task for display purposes.
+     * The format includes the status, description, and due date/time.
+     *
+     * @return the string representation of the deadline task
+     */
+    @Override
+    public String toString() {
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("MMM dd yyyy, h:mm a");
```

Comment:
Perhaps you consider making this into a static final constant, e.g. PREFERRED_DATE_TIME_FORMAT in the superclass Task, such that both Event and Deadline task can use it. This will ensure that the format you want is standardized across your package.